File: static/code/sample/tethered/gdl-acrelayp16-a.cs
// In this sample:
// Continuously switch relay on/off every 2 seconds

using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

void Set(int relay, bool value) {
    var v = value ? 1 : 0;
    duelink.Engine.ExecuteCommand($"Set({relay},{v})");
}

while (true) {
    Thread.Sleep(2000);
    Set(2,true);

    Thread.Sleep(2000);
    Set(2, false);
}

File: static/code/sample/tethered/gdl-acrelayp16-a.py
# In this sample:
# Continuously switch relay 2 on/off every 2 seconds

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
def Set(relay, value):
    v = 1 if value else 0
    duelink.Engine.ExecuteCommand(f"Set({relay},{v})")

while True:
    time.sleep(2)
    Set(2, True)

    time.sleep(2)
    Set(2, False)
